pkg/providers/imagefamily/bootstrap/parser_helpers.go

```diff
@@ -0,0 +1,684 @@
+	}
+
+	return base64.StdEncoding.EncodeToString([]byte(containerdConfig))
+}
+
+func containerdConfigFromNodeBootstrapContract(nbcontract *nbcontractv1.Configuration) (string, error) {
+	if nbcontract == nil {
+		return "", fmt.Errorf("node bootstrap contract is nil")
+	}
+
+	var buffer bytes.Buffer
+	if err := containerdConfigTemplate.Execute(&buffer, nbcontract); err != nil {
+		return "", fmt.Errorf("error executing containerd config template for NBContract: %w", err)
+	}
+
+	return buffer.String(), nil
+}
+
+func getIsMIGNode(gpuInstanceProfile string) bool {
+	return gpuInstanceProfile != ""
+}
+
+func getCustomCACertsStatus(customCACerts []string) bool {
+	return len(customCACerts) > 0
+}
+
+func getEnableTLSBootstrap(bootstrapConfig *nbcontractv1.TLSBootstrappingConfig) bool {
+	return bootstrapConfig.GetTlsBootstrappingToken() != ""
+}
+
+func getEnableSecureTLSBootstrap(bootstrapConfig *nbcontractv1.TLSBootstrappingConfig) bool {
+	// TODO: Change logic to default to false once Secure TLS Bootstrapping is complete
+	return bootstrapConfig.GetEnableSecureTlsBootstrapping()
+}
+
+func getTLSBootstrapToken(bootstrapConfig *nbcontractv1.TLSBootstrappingConfig) string {
+	return bootstrapConfig.GetTlsBootstrappingToken()
+}
+
+func getCustomSecureTLSBootstrapAADServerAppID(bootstrapConfig *nbcontractv1.TLSBootstrappingConfig) string {
+	return bootstrapConfig.GetCustomSecureTlsBootstrappingAppserverAppid()
+}
+
+func getIsKrustlet(wr nbcontractv1.WorkloadRuntime) bool {
+	return wr == nbcontractv1.WorkloadRuntime_WASM_WASI
+}
+
+func getEnsureNoDupePromiscuousBridge(nc *nbcontractv1.NetworkConfig) bool {
+	return nc.GetNetworkPlugin() == nbcontractv1.NetworkPlugin_NP_KUBENET && nc.GetNetworkPolicy() != nbcontractv1.NetworkPolicy_NPO_CALICO
+}
+
+func getHasSearchDomain(csd *nbcontractv1.CustomSearchDomainConfig) bool {
+	if csd.GetDomainName() != "" && csd.GetRealmUser() != "" && csd.GetRealmPassword() != "" {
+		return true
+	}
+	return false
+}
+
+func getCSEHelpersFilepath() string {
+	return cseHelpersScriptFilepath
+}
+
+func getCSEDistroHelpersFilepath() string {
+	return cseHelpersScriptDistroFilepath
+}
+
+func getCSEInstallFilepath() string {
+	return cseInstallScriptFilepath
+}
+
+func getCSEDistroInstallFilepath() string {
+	return cseInstallScriptDistroFilepath
+}
+
+func getCSEConfigFilepath() string {
+	return cseConfigScriptFilepath
+}
+
+func getCustomSearchDomainFilepath() string {
+	return customSearchDomainsCSEScriptFilepath
+}
+
+func getDHCPV6ServiceFilepath() string {
+	return dhcpV6ServiceCSEScriptFilepath
+}
+
+func getDHCPV6ConfigFilepath() string {
+	return dhcpV6ConfigCSEScriptFilepath
+}
+
+// getSysctlContent converts nbcontractv1.SysctlConfig to a string with key=value pairs, with default values.
+//
+//gocyclo:ignore
+//nolint:funlen,gocognit,cyclop // This function is long because it has to handle all the sysctl values.
+func getSysctlContent(s *nbcontractv1.SysctlConfig) string {
```

Comment:
Seems like some of these values and helpers should live in agentbaker and not karpenter.

Reply:
Yes it is, as mentioned in other comments.